types/Cargo.toml

```diff
@@ -12,12 +12,16 @@ license-file = "../LICENSE"
 
 [dependencies]
 base16 = { version = "0.2.1", default-features = false }
+base64 = { version = "0.13.0", default-features = false }
 bitflags = "1"
 blake2 = { version = "0.9.0", default-features = false }
 # TODO: Replace failure with thiserror once no_std support is landed https://github.com/dtolnay/thiserror/pull/64
 datasize = { version = "0.2.0", default-features = false }
+ed25519-dalek = { version = "1.0.0", default-features = false, features = ["rand", "u64_backend"] }
 failure = { version = "0.1.8", default-features = false, features = ["failure_derive"] }
+hex = { version = "0.4.2", default-features = false }
```

Comment:
We can probably avoid importing both `hex` and `base16`, but I'd leave that as a tech debt ticket to keep this already-large PR manageable.